internal/app/machined/pkg/runtime/v1alpha1/platform/gcp/gcp.go

```diff
@@ -71,6 +68,81 @@ func (g *GCP) ParseMetadata(metadata *MetadataConfig) (*runtime.PlatformNetworkC
 		region = region[:idx]
 	}
 
+	publicIPs := []string{}
+
+	for idx, iface := range interfaces {
+		ifname := fmt.Sprintf("eth%d", idx)
+
+		networkConfig.Links = append(networkConfig.Links, network.LinkSpecSpec{
+			Name:        ifname,
+			Up:          true,
+			MTU:         uint32(iface.MTU),
+			ConfigLayer: network.ConfigPlatform,
+		})
+
+		networkConfig.Operators = append(networkConfig.Operators, network.OperatorSpecSpec{
+			Operator: network.OperatorDHCP4,
+			LinkName: ifname,
+			DHCP4: network.DHCP4OperatorSpec{
+				RouteMetric: 1024,
+			},
+			ConfigLayer: network.ConfigPlatform,
+		})
+
+		for _, ipv6addr := range iface.IPv6 {
+			if ipv6addr == "" || iface.GatewayIPv6 == "" {
+				continue
+			}
+
+			ipPrefix, err := netip.ParsePrefix(ipv6addr)
+			if err != nil {
+				return nil, fmt.Errorf("failed to parse ip address: %w", err)
+			}
+
+			if !strings.HasPrefix(ipv6addr, "fd20:") {
+				publicIPs = append(publicIPs, strings.SplitN(ipv6addr, "/", 2)[0])
```

Comment:
we shouldn't expose same address as both assigned to the interface and public IP.

public IP (or external IP how we call it) is only a hack for 1:1 NAT, as the address is not assigned to the interfaces, but it is in fact there via 1:1 NAT. so assigning the address to the interface is all we need

Reply:
Yep, node can have many public accessible IPs (subnets). Only platform knows witch one is for node. 

I think the best way to add all main public accessible IP to the array externalIPs. I noticed that behavior on different platforms. IP exists on the node and meta-server expose it as external IP (1:1 NAT)

Reply:
Hold on, IPv6 addresses are all assigned to the interface (e.g. `eth0`). They _don't_ need to be exposed as `publicIPs`, as they're already on the node.

IPv4 addresses are different - private IP is assigned to the interface, while public IP is not known to the node, so we put as public IP to issue proper certs. 

But there's no reason to do that for IPv6

Reply:
yep, IPv6 have to exist on the node...
I've used externalIP to understand witch one is real(main) ipv6 on the node. And it should be accessible from the internet.

I've removed this code.